tests/zfs-tests/cmd/clonefile.c

```diff
@@ -220,8 +221,8 @@ main(int argc, char **argv)
 		    strcmp(argv[optind+4], "all") == 0) {
 			len = SSIZE_MAX;
 		} else {
-			len = strtoull(argv[optind+4], NULL, 10);
-			if (len == ULLONG_MAX) {
+			len2 = strtoull(argv[optind+4], NULL, 10);
+			if (len2 == ULLONG_MAX) {
```

Comment:
It seems you've assigned the new variable, but never used it.  I guess that is why clone tests are unhappy.

Reply:
@amotin I have updated my patch. Now if len2 is lower then SSIZE_MAX, set len to len2, otherwise use SSIZE_MAX.